From 6cd1cd4d3666b9e36831804c6170b3cf6e376d76 Mon Sep 17 00:00:00 2001
From: Patrick Griffis <tingping@tingping.se>
Date: Thu, 1 Feb 2018 22:07:31 -0500
Subject: [PATCH] Use XDG_RUNTIME_DIR for default socket path

This works in Flatpak by default and is just the better dir.
---
 gnome-screencast.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/gnome-screencast.c b/gnome-screencast.c
index 417ff37..23276ff 100644
--- a/gnome-screencast.c
+++ b/gnome-screencast.c
@@ -299,8 +299,10 @@ static void destroy(void* data)
 
 static void get_defaults(obs_data_t* settings)
 {
+	g_autofree char *default_socket_path = g_build_filename(g_get_user_runtime_dir(), "obs-gnome-screencast",
+																													"shm.sock", NULL);
 	obs_data_set_default_int(settings, "screen", 0);
-	obs_data_set_default_string(settings, "shm_socket", "/tmp/obs-gnome-screencast.sock");
+	obs_data_set_default_string(settings, "shm_socket", default_socket_path);
 	obs_data_set_default_bool(settings, "show_cursor", true);
 	obs_data_set_default_int(settings, "frame_rate", 30);
 }
-- 
2.14.3

